src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/jose/ValidateProofs.kt

```diff
@@ -36,34 +39,32 @@ internal class ValidateProofs(
     private val extractJwkFromCredentialKey: ExtractJwkFromCredentialKey,
 ) {
 
-    context(Raise<InvalidProof>)
     suspend operator fun invoke(
         unvalidatedProofs: NonEmptyList<UnvalidatedProof>,
         credentialConfiguration: CredentialConfiguration,
         at: Instant,
-    ): NonEmptyList<JWK> = coroutineScope {
-        val credentialKeysAndCNonces = unvalidatedProofs.map {
-            when (it) {
-                is UnvalidatedProof.Jwt -> async {
-                    withContext(Dispatchers.Default) {
-                        validateJwtProof(it, credentialConfiguration)
-                    }
+    ): Either<InvalidProof, NonEmptyList<JWK>> = coroutineScope {
+        either {
+            val credentialKeysAndCNonces = unvalidatedProofs.map {
+                when (it) {
+                    is UnvalidatedProof.Jwt ->
+                        withContext(Dispatchers.Default) { validateJwtProof(it, credentialConfiguration).bind() }
```

Comment:
@babisRoutis If we remove `async` we might as well remove `withContext` and `coroutineScope` as well.